### PR TITLE
Add `.vs` folder to `.gitignore`

### PR DESCRIPTION
**Pull request by Mrxx99**
This adds the .vs folder of Visual Studio to .gitignore. Without this it is not possible to switch branches inside Visual Studio as there are always changes (e.g. when only opening a file), but these are locked by Visual Studio.

I also added the .gitignore file to the "Solution Items" folder of the Solution

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@
 *.userosscache
 *.sln.docstates
 .idea/
+.vs
```
Comment (Tyrrrz) [inline .gitignore]:
```suggestion
.vs/
```

**Comment by Tyrrrz** [comment]
Thanks! I discarded changes to `YoutubeDownloader.sln` because it seems to involve things irrelevant to updating the `.gitignore` file.

**Comment by Mrxx99** [comment]
The changes in sln was adding it to the Solution items folder to be able to edit easily in Visual Studio

**Comment by Tyrrrz** [comment]
> The changes in sln was adding it to the Solution items folder to be able to edit easily in Visual Studio

It's fine, I prefer to keep it outside the solution anyway.

**Pull request merged**